Automattic/newspack-popups
# Refactor/segments relationship

### All Submissions:






### Changes proposed in this Pull Request:



Refactors how the Segments and Prompts relationship is stored.

Removes the `selected_segment_id` option and uses the native post - taxonomy relationship

### How to test the changes in this Pull Request:

Make sure you have https://github.com/Automattic/newspack-plugin/pull/2583  and https://github.com/Automattic/newspack-blocks/pull/1498 checked out and built.

We need to test the whole Popups and segmentation feature, from all its angles.

Here is a non-exhaustive list of things to test:

* On a site with many existing prompts and segments, confirm the data migration works fine and they all still work the same way



* on a fresh site, test creating segments and prompts
* test assigning prompts to Segments via the default taxonomy UI in the prompt Editor
* try editing segments, enabling and disabling them, both from the segment edit page and from the toggle in the segments list
* try editing prompts via the Settings Modal in the Campaigns wizard. Try changing the prompts segments
* Confirm that the Conflict warning for uncategorized prompts in the same segment still work as expected
* test the export CLI command
* test the import CLI command
* on a fresh site, test the RAS wizard and confirm the default prompts and segments are created well

Confirm that the `data-segments` html attributes of the popups are correctly filled

Confirm that the Event names for GA tracking gets the correct segment names for the prompt (Search for "Newspack Announcement" in the page source)

* Important: This is basically all I tested. Please think of things I didn't test. We really need to stress it.

### Other information:
